library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -503,11 +533,14 @@ public void onMapReady(MapboxMap mapboxMap) {
                     mapboxMap.getUiSettings().setCompassEnabled(false);
                     // This disables
                     addOnScrollListenerToMap(mapboxMap);
-                    if (droppedPoints != null) {
+                    if (KujakuMapView.this.droppedPoints != null) {
+                        List<io.ona.kujaku.domain.Point> droppedPoints = new ArrayList<>(KujakuMapView.this.droppedPoints);
                         for (io.ona.kujaku.domain.Point point : droppedPoints) {
                             dropPointOnMap(new LatLng(point.getLat(), point.getLng()));
                         }
                     }
+                    mapboxMap.addSource(getPrimaryGeoJsonSource());
+                    mapboxMap.addLayer(getPrimaryLayer());
                 }
```

Comment:
Check if the layer is set before adding layer and geoJsonSource

Reply:
Resolved.